### DISCUSSION TITLE
How to set up a multi-user Windows and Rustdesk firewall

### DISCUSSION THREAD

**Discussion by jackadam1981** [rustdesk/rustdesk#8058] [Q&A]
How to use the% lockappdata% environment variable to set up a firewall?

I started using Rustdesk software in the domain for remote assistance, maintenance and diagnostic work.

This software is free of installation, but it will run in c: \ users \ username \ appdata \ local \ rustdesk \ rustdesk.exe. There are also other files in this directory.

It uses random ports and coordinates distribution through a central server.

I have created a firewall in the domain control that allows% localappdata% \ trustdesk \ trustdesk.exe, all of which are allowed

However, the user end will still pop up firewall requests, with a path of c: \ users \ username \ appdata \ local \ rustdesk \ rustdesk.exe

If the firewall request is cancelled, the connection cannot be made and two rejected firewall policies are automatically created.

Each user will copy rustdesk.exe in their own app data. Do I need to create separate firewalls for each user?

**Comment by DMSnz** [comment]
We decided to deploy via GPO in our domain.
Using a powershell script we install RustDesk and set firewall rules, therefore multi user no drama.
The script I used was largely based on the documentation about install RustDesk via TRMM if you look that up as a starting point.

**Reply by jackadam1981** [reply]
Thank you.
I tried the bat script, and then I tried the powershell script.
Not all terminals here require the use of Rustdesk, so I want to find a more flexible option for installation.
Use tools like Runas to install rights when needed.

**Reply by DMSnz** [reply]
Our method via GPO can obviously be linked to a specific user / device group which could limit who gets it or not, but in our case it was pushed to all so bit easier I guess.
I did have to muck around powershell script quite a lot, but also had to run it as a computer level policy, so it ran as system at logon to install service correctly etc.

**Reply by jackadam1981** [reply]
At this stage, I have self compiled a client with built-in connection parameters and implemented -- silent install using runas and Winrar self decompression, which will be installed in a fixed directory instead of a user directory.
I will try to configure a Group Policy firewall for fixed directories today. It may achieve the goal of flexible installation.

**Reply by DMSnz** [reply]
Sounds like a reasonable plan, I pushed out Firewall rules via GPO also and it seems to have worked.
